vectorstores/chromadb/chroma.go

```diff
@@ -0,0 +1,187 @@
+package chromadb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	chroma "github.com/amikos-tech/chroma-go"
+	chromaopenapi "github.com/amikos-tech/chroma-go/swagger"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+var (
+	// ErrEmbedderWrongNumberVectors is returned when if the embedder returns a number
+	// of vectors that is not equal to the number of documents given.
+	ErrEmbedderWrongNumberVectors = errors.New(
+		"number of vectors from embedder does not match number of documents",
+	)
+	ErrInvalidScoreThreshold = errors.New(
+		"score threshold must be between 0 and 1")
+)
+
+// Store is a wrapper around the chromadb client.
+type Store struct {
+	embedder wrappedEmbeddingFunction
+	client   *chroma.Client
+
+	scheme string
+	host   string
+
+	// optional
+	nameSpaceKey string
+	// optional
+	textKey string
+	// optional: nameSpace represents collection in chromadb
+	nameSpace string
+	// optional
+	distanceFunc chroma.DistanceFunction
+	// optional
+	includes []chroma.QueryEnum
+}
+
+var _ vectorstores.VectorStore = Store{}
+
+// New creates a new Store with options for chromadb.
+func New(opts ...Option) (vectorstores.VectorStore, error) {
+	s, err := applyClientOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := chromaopenapi.NewConfiguration()
+	configuration.Servers = chromaopenapi.ServerConfigurations{
+		{
+			URL:         fmt.Sprintf("%s://%s", s.scheme, s.host),
+			Description: "Chromadb server url for this store",
+		},
+	}
+	s.client = &chroma.Client{
+		ApiClient: chromaopenapi.NewAPIClient(configuration),
+	}
+
+	if _, err = s.client.Heartbeat(); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func (s Store) AddDocuments(ctx context.Context, docs []schema.Document, options ...vectorstores.Option) error {
+	opts := s.getOptions(options...)
+	nameSpace := s.getNameSpace(opts)
+
+	texts := make([]string, 0, len(docs))
+	ids := make([]string, len(docs))
+	for idx, doc := range docs {
+		texts = append(texts, doc.PageContent)
+		ids[idx] = fmt.Sprintf("%d", idx)
+	}
+
+	collection, err := s.client.CreateCollection(s.nameSpace, map[string]interface{}{}, true, s.embedder, s.distanceFunc)
+	if err != nil {
+		return err
+	}
+
+	vectors, err := s.embedder.CreateEmbedding(texts)
+	if err != nil {
+		return err
+	}
+	if len(vectors) != len(texts) {
+		return ErrEmbedderWrongNumberVectors
+	}
+
+	metadatas := make([]map[string]any, 0)
+	for i := 0; i < len(docs); i++ {
+		metadata := make(map[string]any)
+		for k, v := range docs[i].Metadata {
+			metadata[k] = v
+		}
+		metadata[s.textKey] = texts[i]
+		metadata[s.nameSpaceKey] = nameSpace
```

Comment:
This is what I'm planning to do for `NameSpace` (see my lengthy comment above), and later use it as a filter when needed, as the Weaviate adapter does.